pylon/service/api.py

```diff
@@ -3,18 +3,31 @@
 from litestar import Response, get, post, put, status_codes
 
 from pylon._internal.common.endpoints import Endpoint
+from pylon._internal.common.models import Hotkey, Metagraph
 from pylon._internal.common.requests import (
     GenerateCertificateKeypairRequest,
     SetWeightsRequest,
 )
 from pylon._internal.common.settings import settings
+from pylon._internal.common.types import BlockNumber
 from pylon.service.bittensor.client import AbstractBittensorClient
-from pylon.service.bittensor.models import Hotkey
 from pylon.service.tasks import ApplyWeights
 
 logger = logging.getLogger(__name__)
 
 
+@get(Endpoint.METAGRAPH)
+async def get_metagraph(bt_client: AbstractBittensorClient, block_number: BlockNumber | None = None) -> Metagraph:
+    # TODO: TurboBT struggles with fetching old blocks, for tb try to ask for block 4671121
+    if block_number is not None:
+        block = await bt_client.get_block(block_number)
+        if block is None:
+            raise ValueError(f"Block {block_number} not found")
```

Comment:
How `ValueError` translates to a server response? Would it return 400 or 500? Isn't it better to return explicit HTTP response here? 
```suggestion
            return Response(
            {"detail": "Certificate not found or error fetching."}, status_code=status_codes.HTTP_404_NOT_FOUND
            {"detail": f"Block {block_number} not found"}, status_code=status_codes.HTTP_404_NOT_FOUND
        )
        )
```